### PR TITLE
jit: Avoid using mips identifier directly

### PR DESCRIPTION
**Pull request by unknownbrackets**
Quick change for #13739.

-[Unknown]

### CODE REVIEW

File: Core/MIPS/IR/IRInterpreter.cpp [hunk 1]
```diff
@@ -74,167 +74,167 @@ u32 RunMemCheck(u32 pc, u32 addr) {
 }
 
 // We cannot use NEON on ARM32 here until we make it a hard dependency. We can, however, on ARM64.
-u32 IRInterpret(MIPSState *mips, const IRInst *inst, int count) {
+u32 IRInterpret(MIPSState *ms, const IRInst *inst, int count) {
```
Comment (hrydgard) [inline Core/MIPS/IR/IRInterpreter.cpp:77]: is it really necessary to rename mips to ms here and in similar places? Lots of "blame history" pollution and I don't see much benefit

**Review by hrydgard — APPROVED** [review]

**Comment by pcercuei** [comment]
Another trick: add `-Dmips=mips` to the compilation flags to override GCC's macro. Then you won't need to rename variables and it will be more future-proof.

File: Core/MIPS/IR/IRInterpreter.cpp (hunk 1 above)
Comment (hrydgard) [reply Core/MIPS/IR/IRInterpreter.cpp:77]: oh wait, I understand that mips is defined. But can't we just #undef it instead?

**Comment by glebm** [comment]
`undef`ing a compiler-defined macros seems error-prone

**Comment by hrydgard** [comment]
compiler-defined macros with a name as collision-prone as "mips" shouldn't be allowed...

**Comment by glebm** [comment]
I agree with both statements

**Comment by hrydgard** [comment]
just saying, I think it's justified to #undef in this case for that reason.

**Comment by glebm** [comment]
If we `undef` in the header, it might break some GCC stdlib code, depending on header inclusion order.
Might even break in an invisible way if e.g. the code is a MIPS asm optimization with C fallback.

**Comment by pcercuei** [comment]
We can verify that, but I don't think anything depends on that macro in any of the C libraries (since `_mips` and `__mips` are defined too and are much less "dangerous").

File: Core/MIPS/IR/IRInterpreter.cpp (hunk 1 above)
Comment (unknownbrackets) [reply Core/MIPS/IR/IRInterpreter.cpp]: Well, I was worried about `mips` getting depended upon elsewhere, but I think it's pretty safe within this file to undef - I went with the redefine as `mips` approach.

Still kept the changes in other files, but they're mostly just constructors anyway.

-[Unknown]

**Comment by unknownbrackets** [comment]
I think a decent compromise is this in the cpp file:

```c++
#ifdef mips
// Why do MIPS compilers define something so generic?  Try to keep defined, at least...
#undef mips
#define mips mips
#endif
```

Most other files aren't using `mips` alone much, so I kept those changes to `mipsState`.

-[Unknown]

**Comment by hrydgard** [comment]
All for that compromise.

**Review by hrydgard — APPROVED** [review]

**Review by glebm — APPROVED** [review]

**Pull request merged**